Jenkinsfile

```diff
@@ -273,6 +275,40 @@ pipeline {
             }
           }
         }
+        stage('pytest tests') {
+          when {
+            beforeAgent true
+            expression { pytest_test == true }
+          }
+          agent { label 'nixos-mayastor-pytest' }
+          stages {
+            stage('checkout') {
+              steps {
+                checkout([
+                  $class: 'GitSCM',
+                  branches: scm.branches,
+                  extensions: scm.extensions.findAll{!(it instanceof jenkins.plugins.git.GitSCMSourceDefaults)} + [[$class: 'CloneOption', noTags: false, reference: '', shallow: false]],
+                  userRemoteConfigs: scm.userRemoteConfigs
+                ])
+              }
+            }
+            stage('build') {
+              steps {
+                sh 'nix-shell --run "cargo build --bins"'
```

Comment:
are all the binaries in the project needed for the pytests?

Reply:
Certainly not all artifacts are required. However the docker-compose stuff currently requires access to a "working tree". It should be reasonably straight forward to determine which artifacts are required and then copy them to the new agent. I believe the brief is to get these tests working in CI now so that we have as much test coverage as we can as soon as we can and worry about efficiency later. For now the simplest approach (ie. which doesn't require significant changes to the existing code) is to perform another checkout + build on the "pytest" agent.